lib/file/widgets/file_browser.dart

```diff
@@ -72,61 +73,56 @@ class FileBrowser extends StatelessWidget {
-    return FancyCachedBuilder<List<Id<File>>>(
-      controller: path.files.controller,
-      builder: (context, fileIds, isFetching) {
-        logger.w('Files were updated to $fileIds.');
-        if (fileIds?.isEmpty ?? true) {
-          return _buildEmptyState(context);
-        }
-        return FileList(
-          fileIds,
-          primary: false,
-          onOpenDirectory: (directory) => _openDirectory(context, directory),
-          onDownloadFile: (file) => _downloadFile(context, file),
-        );
-      },
+    return CollectionBuilder<File>(
+      collection: path.files,
+      builder: handleEdgeCases(handleEmptyState(
+        emptyStateBuilder: _buildEmptyState,
+        builder: (context, fileIds, _) {
+          return FileList(
+            fileIds,
+            primary: false,
+            onOpenDirectory: (directory) => _openDirectory(context, directory),
+            onDownloadFile: (file) => _downloadFile(context, file),
+          );
+        },
+      )),
     );
   }
 
   Widget _buildStandalone(BuildContext context) {
-    FileBrowserAppBar buildLoadingErrorAppBar(
-      dynamic error, [
-      Color backgroundColor,
-    ]) {
-      return FileBrowserAppBar(
-        title: error?.toString() ?? context.s.general_loading,
-        backgroundColor: backgroundColor,
-      );
-    }
-
     Widget appBar;
     if (isOwnerCourse) {
-      appBar = FancyCachedBuilder<Course>.handleLoading(
-        controller: path.ownerId.controller,
-        builder: (context, course, _) {
+      appBar = EntityBuilder<Course>(
+        id: path.ownerId,
+        builder: (context, snapshot, fetch) {
+          final course = snapshot.data;
+
           if (path.parentId == null) {
             return FileBrowserAppBar(
-              title: course.name,
-              backgroundColor: course.color,
+              title: course?.name ??
+                  snapshot.error?.toString() ??
+                  context.s.general_loading,
+              backgroundColor: course?.color,
             );
           }
 
-          return FancyCachedBuilder<File>.handleLoading(
-            controller: path.parentId.controller,
-            builder: (context, parentDirectory, _) {
+          return EntityBuilder<File>(
+            id: path.parentId,
+            builder: handleEdgeCases((context, parentDirectory, fetch) {
               return FileBrowserAppBar(
                 title: parentDirectory.name,
-                backgroundColor: course.color,
+                backgroundColor: course?.color ?? Colors.white,
               );
-            },
+            }),
           );
         },
       );
     } else if (path.parentId != null) {
-      appBar = FancyCachedBuilder<File>.handleLoading(
-        controller: path.parentId.controller,
-        builder: (context, parent, _) => FileBrowserAppBar(title: parent.name),
+      appBar = EntityBuilder<File>(
+        id: path.parentId,
+        builder: handleEdgeCases(
```

Comment:
We should handle edge cases manually as the default handling doesn't fit an `AppBar`